### PR TITLE
#176 - Decouple Cordova

### PR DESCRIPTION
**Pull request by maverickmishra**
Added code for making addition of browser platform possible through pg- cli

### CODE REVIEW

File: lib/cli/serve.js [hunk 1]
```diff
@@ -3,13 +3,17 @@
  */
 
 var phonegap = require('../main'),
-    console = require('./util/console');
+    console = require('./util/console'),
+    PhoneGap = require('../phonegap'),
+    fs      = require('fs'),
+    path    = require('path');
```
Comment (stevengill) [inline lib/cli/serve.js]: this should end with a comma, not a semi-colon

File: lib/cli/serve.js [hunk 2]
```diff
@@ -42,9 +46,45 @@ module.exports = function(argv, callback) {
...
+    if (!fs.existsSync(path.join(process.cwd(), 'platforms/browser'))) {
+
+       // console.log('in add browser');
+        //console.log(argv);
+        var options = {};
+        options.cmd = 'cordova platform add browser';
+
+        phonegap.cordova(options, function(){
+            
+            gaze('./www/**/*', function(err, watcher) {
```
Comment (stevengill) [inline lib/cli/serve.js]: your indenting is all messed up. Too many unnecessary new lines

File: lib/cli/serve.js [hunk 3]
```diff
@@ -55,17 +79,17 @@ module.exports = function(argv, callback) {
...
         callback(e);
     });
+}
 
-    // gracefully exit on ctrl-c
-    process.on('SIGINT', function() {
-        process.exit();
-    });
-};
+function addBrowser(argv, callback) {
```
Comment (surajpindoria) [inline lib/cli/serve.js]: Can we remove `argv` as a parameter? It's not getting used in the function.

Comment (maverickmishra) [inline lib/cli/serve.js]: Right ! Not using it. Will push in the change soon

File: lib/cli/cordova.js [hunk 4]
```diff
@@ -36,6 +36,13 @@ module.exports = function(argv, callback) {
     
     // stop cordova-cli from tracking us!
     cordovaCommand.push("--no-telemetry");
+    if(cordovaCommand[0] == "cordova" && 
+       cordovaCommand[1] == "plugin" && 
+       cordovaCommand.indexOf("--save") == -1 && cordovaCommand.indexOf("--no-save") == -1 &&
+       cordovaCommand[2] == "add"  || cordovaCommand[2] == "remove" || cordovaCommand[2] == "rm")
+    {
```
Comment (surajpindoria) [inline lib/cli/cordova.js]: When we auto save the platform we do a regex of the command to check what it is, I believe that will work better and help to simplify this conditional.

https://github.com/phonegap/phonegap-cli/blob/master/lib/phonegap/cordova.js#L207

Comment (purplecabbage) [inline lib/cli/cordova.js]: plugin|plugins

Maybe change 
`cordovaCommand[2] == "add"  || cordovaCommand[2] == "remove" || cordovaCommand[2] == "rm")` 
into 
`['add','remove','rm'].indexOf(cordovaCommand[2]) > -1 `

I personally prefer this to regex, this is readable ...

**Comment by camcam1993** [comment]
Need playstore  back

File: lib/phonegap/cordova.js [hunk 5]
```diff
@@ -262,7 +268,9 @@ CordovaCommand.prototype.addPhoneGapJS = function(options, callback) {
     var self = this;
 
     var match = options.cmd.match(/(prepare|compile|build|run|emulate)[ ]*([\w ]*)/);
+
     if (match) {
+        console.log(match);
```
Comment (surajpindoria) [inline lib/phonegap/cordova.js]: Is this `console.log` supposed to be here? If so, we should use the proper method: `self.phonegap.emit('log', match);`

File: spec/cli/create.spec.js [hunk 6]
```diff
@@ -278,29 +279,30 @@ describe('phonegap create <path>', function() {
...
+    //             id: undefined,
+    //             name: undefined,
+    //             config: {},
+    //             'copy-from': undefined,
+    //             'link-to': 'path/to/an-app',
+    //             'verbose': false
+    //         },
+    //         jasmine.any(Function));
+    //     });
+    // });
```
Comment (surajpindoria) [inline spec/cli/create.spec.js]: Do we still need this test here if it is all commented out?

File: lib/cordova/index.js [hunk 7]
```diff
@@ -29,7 +48,7 @@ module.exports.lib.binname = require(path.join(__dirname, '..', '..', 'package.j
  */
 
 module.exports.util = {
-    isCordova: require('cordova').cordova_lib.cordova.findProjectRoot,
+    isCordova: require('cordova-common').CordovaCheck.findProjectRoot,
```
Comment (stevengill) [inline lib/cordova/index.js]: Doesn't look like cordova-common has the cordovaCheck function

**Pull request closed without merging**